File: String/3163.String_Compression_III/C++/3163.String_Compression_III.cpp
class Solution {
public:
    string compressedString(string word) 
    {
        //============================//
        // Input type:                //
        //   - string (word string)   //
        // Return type:               //
        //   - string (result string) //
        //============================//

        /*Initialize*/
        ///// Length of word string /////
        int len_word = word.size();

        ///// Record traversal pointer /////
        int record_trav_ptr = 0;

        ///// Result string /////
        string res_str;


        /*Pointer-based loop traversal*/
        while (record_trav_ptr < len_word) //Whole
        {
            int record_cnter = 1; //Record summary counter

            while (((record_trav_ptr + 1) < len_word) && (word[record_trav_ptr] == word[(record_trav_ptr + 1)])) //Part
            {
                (record_cnter++); (record_trav_ptr++); //Keep updating, traversing

                ///// Check if the current summary counter is larger than 9 or not /////
                if (record_cnter >= 9) { break; }

            } //Part

            res_str += (to_string(record_cnter) + word[record_trav_ptr]); //Keep updating, recording

            (record_trav_ptr++); //Keep updating, traversing

        } //Whole

        return res_str;
    }
};

File: String/3163.String_Compression_III/Python/3163.String_Compression_III.py
class Solution(object):
    def compressedString(self, word):
        """
        :type word: str
        :rtype: str
        """
        #================================#
        # Pointer-based traversal method #
        #================================#

        ############
        #Initialize
        ##### Length of word array #####
        len_word = len(word)

        ##### Record traversal pointer #####
        record_trav_ptr = 0

        ##### Result string #####
        res_str = ""


        ##############################
        #Pointer-based loop traversal
        while (record_trav_ptr < len_word):
            record_cnter = 1 #Record summary counter

            while (((record_trav_ptr + 1) < len_word) and (word[record_trav_ptr] == word[(record_trav_ptr + 1)])):
                record_cnter += 1 #Keep updating/accumulating

                record_trav_ptr += 1 #Keep updating/traversing

                ##### Check if the current summary counter is larger than 9 or not #####
                if (record_cnter >= 9):
                    break

            res_str += (str(record_cnter) + word[record_trav_ptr]) #Keep updating/recording

            record_trav_ptr += 1 #Keep updating/traversing

        return res_str
